### DISCUSSION TITLE
Bluetooth fast connection

### DISCUSSION THREAD

**Discussion by mauroForlimpopoli** [Moddable-OpenSource/moddable#1073] [Ideas]
Good morning,
to save time and so energy in my battery application, I wish to connect my bluetooth slave to master excluding the "discovery time" directly to address.
Is it possible ?
In the "bleclient.js" seams that is enough only 2 parameter to get connection
```
	connect(params) {

		this._connect(params.address, params.addressType);

	}
```
so I tried to connect with 
```
		let buffAdd = new ArrayBuffer(8);
		buffAdd[0]= 0xFC;
		buffAdd[1]= 0xF5;
		buffAdd[2]= 0xC4;
		buffAdd[3]= 0x61;
		buffAdd[4]= 0xA5;
		buffAdd[5]= 0xBA;
		let tt = 0
		this.connect(buffAdd,tt);
```
on the "onDisconnected" function
but I always get "no buffer instance !!"
Later I saw in the xbug tool how is build the buffer, but I can't duplicate it
Can you help me?
regards

**Comment by phoddie** [comment]
The BLE client [connect() method](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/network/ble/ble.md#connectdevice) takes a single argument, a [device object](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/network/ble/ble.md#properties):

```js
connect(params)
```

The connect method expects that the `address` and `addressType` properties are set on the device object. The address must be an ArrayBuffer. That is usually created that with the `address` tagged template. You example above has several problems:

- There is no device object. Instead it passes two arguments.
- The contents of the ArrayBuffer are not initialized with the address. JavaScript does not support `buffer[0] = 1` for ArrayBuffer. You need to use a view to access the buffer contents (`Uint8Array.of(1,2,3).buffer)`). 

You might try reworking your example as:

```js
import GAP from "gap";
import {address} from "btutils";

this.connect({
    address: address`FC:F5:C3:61:A5:BA`,
    addressType: GAP.AddressType.Public
});
```

**Reply by mauroForlimpopoli** [reply]
Thank you Peter,
all works 
Regards

**Reply by phoddie** [reply]
Great! I'll close this one as resolved.
